File: 24. Run Length Encoding - GFG/24.-run-length-encoding.py
#Your task is to complete this function
#Function should return complete string
def encode(arr):
    # Code here
    count = 1
    res = ""
    
    for i in range(len(arr)-1):
        if arr[i] == arr[i+1]:
            count += 1
        else:
            res += str(arr[i]) + str(count)
            count = 1
    
    res += arr[len(arr)-1] + str(count)
    return res
    

#{ 
#  Driver Code Starts
#Your code will prepended here
if __name__=='__main__':
    t=int(input())
    for i in range(t):
        arr=input().strip()
        print (encode(arr))
# } Driver Code Ends

File: 24. Run Length Encoding - GFG/24.-run-length-encoding.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

string encode(string src);    
 
int main() {
	
	int T;
	cin>>T;
	while(T--)
	{
		string str;
		cin>>str;
		
		cout<<encode(str)<<endl;
	}
	return 0;
}// } Driver Code Ends


/*You are required to complete this function */

string encode(string src){     
  //Your code here 
  int i = 0, count = 1;
  string res = "";
  
  for ( ; i < src.length() - 1; i++){
    if(src[i] == src[i+1]){                   // if same character follows
        count++;                              // increase its count
    }  
    else{                                     // else reset the count to 1    
        res = res + src[i] + to_string(count);  // for the new character
        count = 1;
    }   
  }
  res += src[src.size()-1] + to_string(count);  
  return res;
}
  
 
 
